### PR TITLE
Increase tests coverage for search ordering

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Description of the Change
Unit tests for `ElasticPresss/Feature/SearchOrdering/SearchOrdering` (95.09% lines covered).

### Benefits
Increase test coverage for SearchOrdering class.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues
#1422

### CODE REVIEW

File: tests/php/features/TestSearchOrdering.php [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+		$this->assertArrayHasKey( 'ep-pointer', $messages );
+	}
+
+	public function testOutputFeatureBoxSummary() {
+		ob_start();
+		$this->get_feature()->output_feature_box_summary();
+		$output = ob_get_clean();
+
+		$this->assertNotFalse( strpos( $output, 'Insert specific posts into search results for specific search queries.') );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:92]: Should be able to use `assertContains` here. The PHPUnit docs say it's for arrays, but it also works for strings.

File: tests/php/features/TestSearchOrdering.php [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+	 * @return weighting sub-feature
+	 */
+	public function get_feature() {
+		return ElasticPress\Features::factory()->get_registered_feature( 'searchordering' );
+	}
+
+	public function testConstruct() {
+		$instance = new \ElasticPress\Feature\SearchOrdering\SearchOrdering();
+		$this->assertEquals( 'searchordering', $instance->slug );
+		$this->assertEquals( 'Custom Search Results', $instance->title );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:70]: Use `assertSame` to verify both the value and the data type. This check here works fine, but it's good practice.

File: tests/php/features/TestSearchOrdering.php [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+	public function testSubmenuFile() {
+		set_current_screen( 'front' );
+
+		$this->assertEquals( 'test_submenu_file', $this->get_feature()->submenu_file( 'test_submenu_file' ) );
+	}
+
+	public function testRegisterPostType() {
+		$this->get_feature()->register_post_type();
+		$post_types = get_post_types();
+		$this->assertTrue( in_array( 'ep-pointer', $post_types ) );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:136]: You can use `assertContains` here

File: tests/php/features/TestSearchOrdering.php [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+		$GLOBALS['post'] = get_post( $pointer_id );
+
+		$localized_data = $this->get_feature()->get_pointer_data_for_localize();
+
+		$this->assertEquals( 2, count( $localized_data ) );
+		$this->assertArrayHasKey( 'pointers', $localized_data );
+		$this->assertArrayHasKey( 'posts', $localized_data );
+		$this->assertEquals( $post_id_1, $localized_data['pointers'][0]['ID'] );
+		$this->assertEquals( $post_id_2, $localized_data['pointers'][1]['ID'] );
+		$this->assertTrue( $localized_data['posts'][$post_id_1] instanceof \WP_Post );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:177]: There's `assertInstanceOf` that can be used for checking instance types.

File: tests/php/features/TestSearchOrdering.php [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+		$this->assertTrue( in_array( 'findme', wp_list_pluck( get_the_terms( $post_id_1, 'ep_custom_result' ), 'name' ) ) );
+		$this->assertFalse( get_the_terms( $post_id_3, 'ep_custom_result' ) );
+	}
+
+	public function create_term_failed() {
+		return new \WP_Error( 'test_error' );
+	}
+
+	public function testCreateTermFailed() {
+		add_filter( 'pre_insert_term', [ $this, 'create_term_failed'] );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:304]: You can also create a function on-the-fly for creating callbacks if that callback is only used for one unit test:
```
$callback = function() {
   return new \WP_Error( 'test_error' )
};

add_filter( 'pre_insert_term', $callback );
```

File: tests/php/features/TestSearchOrdering.php [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+		remove_filter( 'rest_url', [ $this, 'filter_rest_url_for_leading_slash' ], 10, 2 );
+	}
+
+	public function filter_rest_url_for_leading_slash( $url, $path ) {
+		if ( is_multisite() || get_option( 'permalink_structure' ) ) {
+			return $url;
+		}
+
+		// Make sure path for rest_url has a leading slash for proper resolution.
+		$this->assertTrue( 0 === strpos( $path, '/' ), 'REST API URL should have a leading slash.' );
```
Comment (petenelson) [inline tests/php/features/TestSearchOrdering.php:436]: `assertStringStartsWith`

**Comment by petenelson** [comment]
I'd recommend updating various `assertFalse` and `assertTrue` statements with more granular asserts, such as `assertContains`, `assertInstanceOf`, etc. This will give better failure messages when an assert fails.

For example, if we had the following test: 
```
$fruit = [
	'banana',
	'apple',
	'orange',
];

// Verify we have no bananas.
$this->assertFalse( in_array( 'banana', $fruit ) );
```
Running this test will give the following message:
`Failed asserting that true is false`

But using a more-appropriate assert gives a better result:
```
// Verify we have no bananas.
$this->assertNotContains( 'banana', $fruit );
```
`Failed asserting that an array does not contain 'banana'.`

Also, please add docblocks for each function, and include a `@group` parameter which allows the unit tests to be run on an entire group of tests, example:
```
./vendor/bin/phpunit --group=feature-search-ordering
```

**Comment by dinhtungdu** [comment]
Thanks so much for the detailed review @petenelson!
I got your point and update my two PRs

**Pull request merged**